manuscript_analyses/preprocess.step2.sh

```diff
@@ -0,0 +1,104 @@
+
+# add in insert size from file Kristen created for me from metadata
+./merge_sets.pl -header -x insert_size_metrics.txt=0,0,2 unfiltered_sample_annotations.annot.txt > tmp
+mv tmp unfiltered_sample_annotations.annot.txt
+
+############################################
+# File all.sample2var.unfiltered.noannot.txt: file has 1 line per sample-variant
+# Add the following 4 annotation columns:
+# 1) hap_defining_variant : blank or hap_defining_variant (if known haplogroup variant)
+# 2) numtID : blank or numtA or numtB based on PacBio validateion
+# 3) release3.1.1 : blank or true (if sample in final release set)
+# 4) mtCN : mtDNA copy number of sample
+#
+# to do this we merge in the colunmns we want from other files
+# note the first column in SAMPLE_ID (with spaces) but need to merge via participant_id (no spaces)
+#
+############################################
+
+cp all.sample2var.unfiltered.noannot.txt tmp1.txt
+cut -f 1 all.sample2var.unfiltered.noannot.txt > t1
+./merge_sets.pl -header -byname -x unfiltered_sample_annotations.annot.txt=SAMPLE_ID,s,participant_id t1 | cut -f 2 > t1.remap
+cut -f 2-5 all.sample2var.unfiltered.noannot.txt > t2
+paste t1.remap t2 > tmp1.txt
+./merge_sets.pl -header -byname -x ./summary.var.txt=POS.REF.ALT,POS.REF.ALT,hap_defining_variant tmp1.txt > tmp2.txt
+mv tmp2.txt tmp1.txt
+./merge_sets.pl -header -byname -x numtAB.txt=POS.REF.ALT,variant,numtID tmp1.txt > tmp2.txt
+mv tmp2.txt tmp1.txt
+./merge_sets.pl -header -byname -x tmp.release_samples.txt=participant_id,participant_id,release3.1.1 tmp1.txt > tmp2.txt
+mv tmp2.txt tmp1.txt
+./merge_sets.pl -header -byname -x unfiltered_sample_annotations.txt=participant_id,participant_id,mtCN tmp1.txt > all.sample2var.unfiltered.txt
+
+############################################
+# Now create passplus 
+# which ignores these filters: contamination|low_allele_frac|mt_many_low_hets|possible_numt
+# but gets rid of other FAIL filters
+############################################
+
+cat all.sample2var.unfiltered.txt | grep -v -P "(FAIL|base_qual|position|strand_bias|weak_evidence|black)" > all.sample2var.passplus.txt
+
+################################################
+# convert vep_unfiltered.vcf into tab-del summary.unfiltered.var.vep.txt
+############################################
+grep -v "^\#\#" vep_unfiltered.vcf | cut -f 1-8 | perl -p -e 's/^(\S+)\t(\S+)\t(\S+)\t(\S+)\t(\S+)\t(\S+)\t(\S+)\t/$2\.$4\.$5\t$7\t/g' | perl -p -e 's/\t[^\t]*CSQ=([^;]+).*$/\t$1/g' | perl -p -e 's/INFO/Allele|Consequence|IMPACT|SYMBOL|Gene|Feature_type|Feature|BIOTYPE|EXON|INTRON|HGVSc|HGVSp|cDNA_position|CDS_position|Protein_position|Amino_acids|Codons|ALLELE_NUM|DISTANCE|STRAND|VARIANT_CLASS|MINIMISED|SYMBOL_SOURCE|HGNC_ID|CANONICAL|TSL|APPRIS|CCDS|ENSP|SWISSPROT|TREMBL|UNIPARC|GENE_PHENO|SIFT|PolyPhen|DOMAINS|HGVS_OFFSET|MOTIF_NAME|MOTIF_POS|HIGH_INF_POS|MOTIF_SCORE_CHANGE|LoF|LoF_filter|LoF_flags|LoF_info/g' | perl -p -e 's/\|/\t/g' | cut -f 1-10 > summary.unfiltered.var.vep.txt
```

Comment:
You gave me a version of the file with the CSQ VEP annotations in it.  It might not have been the most recent version, but I checked and they were the same variants.  But this does make it non-reproduceable.  Can you generate a file with the VEP annotations from the most recent unfiltered VCF?

Reply:
the most recent one is the one posted in May:
https://the-tgg.slack.com/archives/C58J03659/p1620069937094200?thread_ts=1619539539.091100&cid=C58J03659

Reply:
can this be marked as resolved?